optimize (u)int8 vectorized operator*

mikekgfb opened the pull request:
Summary: optimize (u)int8 vectorized operator*

Test Plan: sandcastle github

Differential Revision: D52318192




cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

jgong5 reviewed (commented):
LGTM. Can you add avx512 counterpart too?

mikekgfb commented:
> LGTM. Can you add avx512 counterpart too?

Submitted as #116346

mikekgfb commented:
@pytorchbot merge

malfet commented on aten/src/ATen/cpu/vec/vec256/vec256_int.h:
```diff
@@ -960,13 +960,37 @@ Vectorized<T> inline int_elementwise_binary_256(const Vectorized<T>& a, const Ve
 template <>
 Vectorized<int8_t> inline operator*(const Vectorized<int8_t>& a, const Vectorized<int8_t>& b) {
   // We don't have an instruction for multiplying int8_t
```
```suggestion
  // We don't have an instruction for multiplying int8_t, so upcast to int16_t and then mask
```

mikekgfb commented:
@pytorchbot merge

mikekgfb commented:
@pytorchbot merge

Commit:
optimize (u)int8 vectorized operator* (#116235)

Summary:

optimize (u)int8 vectorized operator*

Test Plan: sandcastle github

Reviewed By: malfet, hl475

Differential Revision: D52318192

mikekgfb commented:
@pytorchbot merge
